### ISSUE TITLE
Improve accessibility of card link titles

### ISSUE THREAD

**Issue by danahertzberg** [az-digital/az_quickstart#3793]
## Problem/Motivation
Issue found on a [Dubbot scan](https://app.dubbot.com/sites/5dc32d02482faf0915d31627/pages/6636d6ca337a96db153cbe7f?tab=accessibility&issueId=link-name).

### Describe the bug
When the Card Link URL field is filled out and the Card Link Title is _not_ populated, there is an accessibility concern because screen readers cannot tell what the link is.

### To Reproduce
1. Add a card to a page
2. Complete the Card Link URL field
3. Do not fill in the Card Link Title field
4. Save
5. Inspect the page and find the card
6. Observe no title or content around the `<a>` tag on the card

## Proposed resolution
1. Require the Card Link Title field -- Only when URL field is populated
2. Add a selection in the Card Link Style dropdown for `Hidden` or similar

### Expected behavior
I expect links to have accurate descriptive link titles.

## Additional context
Consider `sr-only` class. https://getbootstrap.com/docs/4.0/utilities/screenreaders/

Expect to create a database update to look for empty titles, with populated URLs. In this case, we should have a reasonable default. Maybe auto-populate the link title with the card title and select the hidden link style? 

Or just not allow save and have a notice with help text?

**Comment by danahertzberg** [comment]
Solution includes the following HTML as approved by @accesswatch.

Add a `<span class="sr-only">` around the link title within the `<a>` tag.

**Comment by danahertzberg** [comment]
Need help with the `Require the Card Link Title field -- Only when URL field is populated` task. 🙏

**Comment by danahertzberg** [comment]
Joe added some code to help the requirement needs. Dana to evaluate locally.
